src/towncrier/_project.py

```diff
@@ -37,13 +44,30 @@ def _get_package(package_dir: str, package: str) -> ModuleType:
     return module
 
 
+def _get_metadata_version(package: str) -> str | None:
+    distributions = packages_distributions()
+    distribution_names = distributions.get(package)
+    if not distribution_names or len(distribution_names) != 1:
+        # We can only determine the version if there is exactly one matching distribution.
+        return None
+    return metadata_version(distribution_names[0])
+
+
 def get_version(package_dir: str, package: str) -> str:
+    version: Any
+
+    # First try to get the version from the package metadata.
+    if version := _get_metadata_version(package):
+        return version
+
     module = _get_package(package_dir, package)
 
     version = getattr(module, "__version__", None)
 
     if not version:
-        raise Exception("No __version__, I don't know how else to look")
+        raise Exception(
+            f"No __version__ or metadata version info for the {package} package."
```

Comment:
A minor comment to make the package name explicit


```suggestion
            f'No __version__ or metadata version info for the "{package}" package.'
```

The message `No __version__ or metadata version info for the missing package.` as a bit confusing at first.